23 - Change ownership of var/cache to nginx user

csullivannet opened the pull request:
Resolves: https://github.com/nginxinc/docker-nginx-unprivileged/issues/23

The alpine images fail to start because the `nginx` user has no access to `/var/cache/nginx`.

`-H` must be removed so that the directory is created when instantiating the user.

alessfg commented:
Thanks for the PR @csullivannet! 

Do you know if this issue also applies to the `buster` (previously `stretch`) images? During my testing I didn't run into any issues after I removed `chown` and `chmod` to adjust permissions.

Also, I'd would like to keep the first RUN block identical to the official NGINX Dockerfiles -- so instead of tweaking the `adduser` command I would rather readd `chown` and `chmod`. Do you think you could make the necessary changes?

Cheers,
Alessandro.

csullivannet commented:
I haven't tested `buster` but I can pretty easily check whether they're affected as well. I'll fixup the PR and readd `chown`+`chmod` instead of doing it inline with `adduser`.

csullivannet commented:
I ran a test on the `buster` based images and the issue persists.

csullivannet commented:
@alessfg I've updated the pull request to leave the first `RUN` untouched, and added the `chown`/`chmod` as a separate `RUN` across all `Dockerfile`s. I've tested both the `alpine` and `buster` variants and this resolves the issue for me.

alessfg commented:
Merged -- thanks again for submitting the PR!
